### PR TITLE
feat: Add -i flag to specify ignore files

### LINKED ISSUE #428: Ability to ignore node_modules or files in .gitignore by default
Referencing issue #427, I am working on a project where I'm co-locating tests with their module.

i.e.
```
--/path/to/module.js
--/path/to/module.spec.js
```

I'm currently targeting these spec files with the function `tape '{,!(node_modules)/**/}*.spec.js' | tap-spec`. The glob I've written is somewhat convoluted and slightly difficult to read and it does seem like perhaps the default behaviour for the tape-cli is that it should exclude the `node_modules` directory or respect your `.gitignore` file.

Within the glob module that tape uses, there is an `ignore` option to exclude certain directories and files. If there are legitimate reasons for not excluding node_modules and/or gitignore by default, perhaps this could be achieved behind a flag.

Thoughts?

### PR DESCRIPTION
**Pull request by r0mflip**
Fixes #428 

Confirming to https://github.com/substack/tape/issues/428#issuecomment-367939098

Closes #451.

### CODE REVIEW

File: bin/tape [hunk 1]
```diff
@@ -29,6 +31,15 @@ opts.require.forEach(function (module) {
     }
 });
 
+// Load .gitignore from cwd
+var ignoreStr = '';
+try {
+    ignoreStr = readFileSync(resolvePath(cwd, opts.ignore === '' ? '.gitignore' : opts.ignore), 'UTF-8');
```
Comment (ljharb) [inline bin/tape]: nit:
```suggestion
    ignoreStr = readFileSync(resolvePath(cwd, opts.ignore === '' ? '.gitignore' : opts.ignore), 'utf-8');
```

File: bin/tape [hunk 2]
```diff
@@ -29,6 +31,15 @@ opts.require.forEach(function (module) {
...
+} catch (e) {}
```
Comment (ljharb) [inline bin/tape]: i think this should probably exit with an error if `opts.ignore` is provided and the file read doesn't work?

File: bin/tape [hunk 3]
```diff
@@ -39,6 +50,10 @@ opts._.forEach(function (arg) {
     }
 
     files.forEach(function (file) {
+        if (matcher.shouldIgnore(file)) {
```
Comment (ljharb) [inline bin/tape]: instead of this, let's `files.filter(function (file) { return !matcher.shouldIgnore(file); }).forEach`?

**Comment by r0mflip** [comment]
Sorry I didn't look at any of your comments and review. I am on the other side of the world as you. Timing issues 😝

**Comment by substack** [comment]
This is a really weird feature. tape assumes git now and reads gitignore files? I am very against this.

**Comment by ljharb** [comment]
@substack are you only against the default of `.gitignore`, or against the entire `-i` feature?

**Pull request merged**